### PR TITLE
fix that it didn't allowed repeated comments and/or likes

### PR DESCRIPTION
**Pull request by sebastianibam002**

### CODE REVIEW

File: backend/src/main/java/controllers/MySQLController.java [hunk 1]
```diff
@@ -425,35 +432,47 @@ public boolean addNewUser(User newUser) {
...
+
     private void addLikesPosts(Post[] posts){
         // add all the likes
-        HashMap<String, User> likesHashMap = this.getAllLikes();
+        HashMap<String, ArrayList<User>> likesHashMap = this.getAllLikes();
         for (String postId: likesHashMap.keySet()) {
             for (Post post : posts) {
                 if (post.getId().equals(postId)) {
-                    post.addLike(likesHashMap.get(postId));
+                    for(User user: likesHashMap.get(postId)){
```
Comment (ericjest) [inline backend/src/main/java/controllers/MySQLController.java:473]: space here after for, for style

File: backend/src/main/java/controllers/MySQLController.java [hunk 2]
```diff
@@ -398,20 +398,27 @@ public boolean addNewUser(User newUser) {
                 String userId = likesResult.getString("user_id");
                 // now I just need the post ID
                 String postId = likesResult.getString("post_id");
-                postLike.put(postId, new User(username, password, bio, userId));
+                if (!postLike.containsKey(postId)) {
+                    ArrayList<User> userArray = new ArrayList<User>();
```
Comment (ericjest) [inline backend/src/main/java/controllers/MySQLController.java:402]: I'm a little confused what's happening here, would a more descriptive variable name for userArray help maybe?  And same with "postLike", it sounds like that is one like, but it's a hash map. Can you explain what is happening and if you think it is good idea change the variable names?

**Review by ericjest — CHANGES_REQUESTED** [review]

File: backend/src/main/java/controllers/MySQLController.java [hunk 3]
```diff
@@ -398,20 +398,27 @@ public boolean addNewUser(User newUser) {
...
+            return new HashMap<String, ArrayList<User>>();
         }
     }
 
 
-    private HashMap<String, Comment> getAllComments(){
+    private HashMap<String, ArrayList<Comment>> getAllComments(){
         try {
-            HashMap<String, Comment> postComment = new HashMap<>();
+            HashMap<String, ArrayList<Comment>> postComment = new HashMap<>();
```
Comment (yowang20) [inline backend/src/main/java/controllers/MySQLController.java:421]: Should this variable name be "postComments"?

File: backend/src/main/java/controllers/MySQLController.java (hunk 2 above)
Comment (yowang20) [reply backend/src/main/java/controllers/MySQLController.java:402]: I think it should be "postLikes"?

**Pull request merged**